airflow/providers/databricks/hooks/databricks.py

```diff
@@ -114,6 +116,53 @@ def from_json(cls, data: str) -> RunState:
         return RunState(**json.loads(data))
 
 
+class ClusterState:
+    """Utility class for the cluster state concept of Databricks cluster."""
+
+    CLUSTER_LIFE_CYCLE_STATES = [
+        "PENDING",
+        "RUNNING",
+        "RESTARTING",
+        "RESIZING",
+        "TERMINATING",
+        "TERMINATED",
+        "ERROR",
+        "UNKNOWN",
+    ]
+
+    def __init__(self, state: str = "", state_message: str = "", *args, **kwargs) -> None:
+        self.state = state
+        self.state_message = state_message
+
+    @property
+    def is_terminal(self) -> bool:
+        """True if the current state is a terminal state."""
+        if self.state not in self.CLUSTER_LIFE_CYCLE_STATES:
+            raise AirflowException(f"Unexpected cluster life cycle state: {self.state}")
```

Comment:
could you move this to `__init__`?

Reply:
Ok, but these lines are come from `RunState.is_terminal()` method. Should I also change the `RunState` too?

Reply:
I moved unexpected run/cluster state checking to `__init__()` level. Thanks! 👍